metrics/src/main/java/mobile/aerogear/org/metrics/MetricsService.java

```diff
@@ -0,0 +1,122 @@
+package mobile.aerogear.org.metrics;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.content.pm.PackageManager;
+
+import org.aerogear.mobile.core.MobileCore;
+import org.aerogear.mobile.core.ServiceModule;
+import org.aerogear.mobile.core.configuration.ServiceConfiguration;
+import org.aerogear.mobile.core.http.HttpRequest;
+import org.aerogear.mobile.core.http.HttpResponse;
+import org.aerogear.mobile.core.http.HttpServiceModule;
+import org.aerogear.mobile.core.logging.Logger;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.UUID;
+
+public class MetricsService implements ServiceModule {
+    private final static String MODULE_NAME = "metrics";
+    private final static String LOG_TAG = "AEROGEAR/METRICS";
+    private final static String STORAGE_NAME = "mobile.aerogear.org.metrics";
+    private final static String STORAGE_KEY = "metrics-sdk-installation-id";
+
+    private HttpServiceModule httpService;
+    private MetricsConfig config;
+    private Logger logger;
+
+    /**
+     * Get or create the client ID that identifies a device as long as the user doesn't
+     * reinstall the app or delete the app storage. A random UUID is created and stored in the
+     * application shared preferences.
+     *
+     * @param context Android app context
+     * @return String Client ID
+     */
+    private String getOrCreateClientId(final Context context) {
+        final SharedPreferences preferences = context
+            .getSharedPreferences(STORAGE_NAME, Context.MODE_PRIVATE);
+
+        String clientId = preferences.getString(STORAGE_KEY, null);
+        if (clientId == null) {
+            clientId = UUID.randomUUID().toString();
+
+            logger.info(LOG_TAG, "Generated a new client ID: " + clientId);
+
+            SharedPreferences.Editor editor = preferences.edit();
+            editor.putString(STORAGE_KEY, clientId);
+            editor.commit();
+        }
+
+        return clientId;
+    }
+
+    /**
+     * Get the version of the app itself
+     *
+     * @param context Android application context
+     * @return String version name
+     */
+    private String getAppVersion(final Context context) {
+        try {
+            return context
+                .getPackageManager()
+                .getPackageInfo(context.getPackageName(), 0).versionName;
+        } catch (PackageManager.NameNotFoundException e) {
+            logger.error(LOG_TAG, e);
```

Comment:
Not sure if's going to happen but I think we should raise an Exception instead of return null